Core/Src/Systems/External/Can/Can.c

```diff
@@ -0,0 +1,85 @@
+#include "../../../../Inc/Systems/External/Can/Can.h"
+#include "stdio.h"
+#include <string.h>
+#include <stdlib.h>
+
+int parseCanData(CanMessage* canMsg, const char* fn) {
+    FILE* file = fopen(fn, "r");
```

Comment:
Can add a validation to check with either ```CanMessage``` or ```fn``` is NULL to avoid crashes when writing in file/opening file. 
For example: 
```
if (!canMsg || !fn) {
        return 0;
    }
```